### PR TITLE
MutualGroupDMs: Simplified profiles support

### PR DESCRIPTION
**Pull request by Sqaaakoi**
Screenshot of the Mutual Groups tab in the new simplified profiles UI

### CODE REVIEW

File: src/plugins/mutualGroupDMs/index.tsx [hunk 1]
```diff
@@ -58,9 +59,32 @@ export default definePlugin({
...
+                },
+                {
+                    match: /(?<=(\i)===\i\.UserProfileSections\.MUTUAL_GUILDS?.{0,150}\}\):)/,
+                    replace: '$1==="MUTUAL_GDMS"?$self.renderMutualGDMs(arguments[0]):'
+                },
+            ]
+        },
     ],
 
+    useGDMCount(userId: string) {
```
Comment (Vendicated) [inline src/plugins/mutualGroupDMs/index.tsx:83]: i think making this a hook is fairly pointless

it's not that important to update the count if a new dm gets added and just adds extra complexity. you save yourself one patch if you just don't

File: src/plugins/mutualGroupDMs/index.tsx [hunk 2]
```diff
@@ -58,9 +59,32 @@ export default definePlugin({
...
+                {
+                    match: /(?<=let\{user:(\i),.{0,700})(?=return\(null!=)/,
+                    replace: "let vencordMutualGroupsCount=$self.useGDMCount($1.id);"
+                },
+                {
+                    match: /(?<=(\i\.push)\(\{section:\i\.UserProfileSections\.MUTUAL_GUILDS,text:.{0,250}\}\)\)\}\))/,
+                    replace: ',$1({section:"MUTUAL_GDMS",text:vencordMutualGroupsCount})'
+                },
+                {
+                    match: /(?<=(\i)===\i\.UserProfileSections\.MUTUAL_GUILDS?.{0,150}\}\):)/,
```
Comment (Vendicated) [inline src/plugins/mutualGroupDMs/index.tsx:76]: all of these matches seem like they could cause slow runtime due to the lookbehinds

please check the Tracer timings to see if they are slow (anything under ~2ms) and if yes try to fix them, eg by moving lookbehinds after the normal match or not using lookbehind

**Review by Vendicated — COMMENTED** [review]
thanks for the fix!

File: src/plugins/mutualGroupDMs/index.tsx (hunk 2 above)
Comment (Sqaaakoi) [reply src/plugins/mutualGroupDMs/index.tsx:76]: i've finally gotten around to this, i've tested the patches in power saving mode and the first patch ` /(?<=let\{user:(\i),.{0,700})(?=return\(null!=)/` is super slow (14.5ms)
I will fix this; do you think the `.{0,700}` is the issue?

Comment (Sqaaakoi) [reply src/plugins/mutualGroupDMs/index.tsx:76]: ok I fixed the lag by just splitting it into 2 patches where the first one sets a variable, is this an appropriate solution?

File: src/plugins/mutualGroupDMs/index.tsx (hunk 1 above)
Comment (Sqaaakoi) [reply src/plugins/mutualGroupDMs/index.tsx:87]: it looks like all the other tabs use a hook to calls useStateFromStores to their respective stores anyway

we have 4 options
- leave as-is witb 2 patches
- replace my new hacky variable with the hook variable (i will try this now)
~~- move hook into a condition that is never changed (break the rules of hooks?)~~ obsoleted idea while I was typing this comment
- remove hook

File: src/plugins/mutualGroupDMs/index.tsx (hunk 2 above)
Comment (Sqaaakoi) [reply src/plugins/mutualGroupDMs/index.tsx:80]: forget that see other comment thread

**Review by Sqaaakoi — COMMENTED** [review]
Sqaaakoi-09-06-2024-15-46-59
how will we deal with this? it's starting to clip off the right side of the tab bar, and it will get even worse when ReviewDB (maybe even PermissionsViewer too?) gets updated to work with the simplified profiles

**Comment by Vendicated** [comment]
superseded

**Pull request closed without merging**